frontend/src/store/wallet.ts

```diff
@@ -522,6 +527,22 @@ export default function useWalletStore() {
     return null;
   });
 
+  /**
+   * @notice Copies the address of type to the clipboard
+   */
+  async function copyAddress(address: string) {
+    if (!provider.value) return;
+    const mainAddress = await toAddress(address, provider.value);
+    await copyToClipboard(mainAddress);
+    notifyUser('success', `${tc('WalletRow.address-copied')}`);
+  }
+
+  function openInEtherscan(hash: string) {
```

Comment:
This method and `copyAddress` above feel like they should be in `utils.ts` and take a `provider` input, since they aren't otherwise reliant on the wallet store